src/pages/core/guides/anchor/how-to-create-a-core-collection-with-anchor.md

```diff
@@ -0,0 +1,349 @@
+}
+```
+
+Some accounts in the `CreateCollection` struct are marked as `optional`. This is because, in the definition of the `CreateCollectionV2CpiBuilder`, certain accounts can be omitted.
+
+```rust
+/// ### Accounts:
+///
+///   0. `[writable, signer]` collection
+///   1. `[optional]` update_authority
+///   2. `[writable, signer]` payer
+///   3. `[]` system_program
+```
+
+To make the example as flexible as possible, every `optional` account in the program instruction is also treated as `optional` in the `create_core_collection` instruction's account struct.
+
+### Creating the Instruction
+
+The `create_core_collection` function utilizes the inputs from the `CreateCollection` account struct and the `CreateCollectionArgs` arg struct that we defined earlier to interact with the `CreateCollectionV2CpiBuilder` program instruction.
+
+```rust
+pub fn create_core_collection(ctx: Context<CreateCollection>, args: CreateCollectionArgs) -> Result<()> {
+  let update_authority = match &ctx.accounts.update_authority {
+      Some(update_authority) => Some(update_authority.to_account_info()),
+      None => None,
+  };
+  
+  CreateCollectionV2CpiBuilder::new(&ctx.accounts.mpl_core_program.to_account_info())
+      .collection(&ctx.accounts.collection.to_account_info())
+      .payer(&ctx.accounts.payer.to_account_info())
+      .update_authority(update_authority.as_ref())
+      .system_program(&ctx.accounts.system_program.to_account_info())
+      .name(args.name)
+      .uri(args.uri)
+      .invoke()?;
+
+  Ok(())
+}
+```
+
+In this function, the accounts defined in the `CreateCollection` struct are accessed using `ctx.accounts`. Before passing these accounts to the `CreateCollectionV2CpiBuilder` program instruction, they need to be converted to their raw data form using the `.to_account_info()` method. 
+
+This conversion is necessary because the builder requires the accounts in this format to interact correctly with the Solana runtime.
+
+Some of the accounts in the `CreateAsset` struct are `optional`, meaning their value could be either `Some(account)` or `None`. To handle these optional accounts before passing them to the builder, we use a match statement that allows us to check if an account is present (Some) or absent (None) and based on this check, we bind the account as `Some(account.to_account_info())` if it exists, or as `None` if it doesn't. Like this:
+
+```rust
+let update_authority = match &ctx.accounts.update_authority {
+    Some(update_authority) => Some(update_authority.to_account_info()),
+    None => None,
+};
+```
+
+After preparing all the necessary accounts, we pass them to the `CreateCollectionV2CpiBuilder` and use `.invoke()` to execute the instruction, or `.invoke_signed()` if we need to use signer seeds.
+
+For more details on how the Metaplex CPI Builder works, you can refer to this [documentation](/guides/rust/how-to-cpi-into-a-metaplex-program#using-metaplex-rust-transaction-cpi-builders)
```

Comment:
@tonyboylehub I added it here, do you think we should make it more visible/explain how CPI works or this is fine?

Reply:
I like that. Thanks!